flutter-for-android.md

```diff
@@ -0,0 +1,1845 @@
+and CustomPainter which implements your algorithm to draw to canvas.
+
+In this popular StackOverFlow.com answer you can see how
+a signature painter is implemented.
+
+See https://stackoverflow.com/questions/46241071/create-signature-area-for-mobile-app-in-dart-flutter
+
+<!-- skip -->
+{% prettify dart %}
+import 'package:flutter/material.dart';
+class SignaturePainter extends CustomPainter {
+  SignaturePainter(this.points);
+  final List<Offset> points;
+  void paint(Canvas canvas, Size size) {
+    Paint paint = new Paint()
+      ..color = Colors.black
+      ..strokeCap = StrokeCap.round
+      ..strokeWidth = 5.0;
+    for (int i = 0; i < points.length - 1; i++) {
+      if (points[i] != null && points[i + 1] != null)
+        canvas.drawLine(points[i], points[i + 1], paint);
+    }
+  }
+  bool shouldRepaint(SignaturePainter other) => other.points != points;
+}
+class Signature extends StatefulWidget {
+  SignatureState createState() => new SignatureState();
+}
+class SignatureState extends State<Signature> {
+  List<Offset> _points = <Offset>[];
+  Widget build(BuildContext context) {
+    return new GestureDetector(
+      onPanUpdate: (DragUpdateDetails details) {
+        setState(() {
+          RenderBox referenceBox = context.findRenderObject();
+          Offset localPosition =
+          referenceBox.globalToLocal(details.globalPosition);
+          _points = new List.from(_points)..add(localPosition);
+        });
+      },
+      onPanEnd: (DragEndDetails details) => _points.add(null),
+      child: new CustomPaint(painter: new SignaturePainter(_points)),
+    );
+  }
+}
+class DemoApp extends StatelessWidget {
+  Widget build(BuildContext context) => new Scaffold(body: new Signature());
+}
+void main() => runApp(new MaterialApp(home: new DemoApp()));
+{% endprettify %}
+
+
+**How do I build custom Widgets?**
+
+Building custom widgets in Flutter is as simple as extending a Widget and 
+adding your custom logic. This is like Android where you would extend a View 
+and override the methods to customize it.
+
+Let's take a look at how to build a CustomButton that takes in the label in the 
+constructor and displays it
+
+<!-- skip -->
+{% prettify dart %}
+class CustomButton extends StatelessWidget {
+  final String label;
+  CustomButton(this.label);
+
+  @override
+  Widget build(BuildContext context) {
+    return new RaisedButton(onPressed: (){
+
+    }, child: new Text(label));
+  }
+}
+{% endprettify %}
+
+Then you can use this CustomButton just like you would with any other Widget
+
+<!-- skip -->
+{% prettify dart %}
+@override
+Widget build(BuildContext context) {
+  return  new Center(
+        child: new CustomButton("Hello"),
+  );
+}
+{% endprettify %}
+
+# Intents
+
+**What is the equivalent of an Intent in Flutter?**
+
+Flutter does not have the concept of Intents. In Android, there are two 
+main use-cases for Intents, One is to switch between Activities and another is 
```

Comment:
"In Android, there are two main use cases for Intents: switching between Activities, and invoking external components."

Reply:
done